keps/sig-api-machinery/2876-crd-validation-expression-language/README.md

```diff
@@ -0,0 +1,660 @@
+need to make an assessment of how much support we have in older kubernetes versions for this feature.
+
+###### What specific metrics should inform a rollback?
+
+Custom resource create/update failures.
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+Will be completed before Beta.
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+No
+
+### Monitoring Requirements
+
+<!--
+This section must be completed when targeting beta to a release.
+-->
+
+###### How can an operator determine if the feature is in use by workloads?
+
+Check if there exist any custom resource definition with the x-kubernetes-validator field in the OpenAPIv3 schema.
+
+###### How can someone using this feature know that it is working for their instance?
+
+Test that a validation rule rejects a custom resource create/update/patch/apply as expected.
+
+###### What are the reasonable SLOs (Service Level Objectives) for the enhancement?
+
+No impact on latency for custom resource create/update/patch/apply when validation rules are absent
+from a custom resource definition.
+
+Performance when validation rules are in use will need to be measured and optimized. We anticipate negligible
+impact (<5%) for typical use.
+
+###### What are the SLIs (Service Level Indicators) an operator can use to determine the health of the service?
+
+Custom resource definition create/update/patch/apply latencies are available today and should be sufficient.
+
+###### Are there any missing metrics that would be useful to have to improve observability of this feature?
+
+We don't anticipate the performance implications to justify the introduction of a validation latency metric, but
+if performance 
+
+### Dependencies
+
+###### Does this feature depend on any specific services running in the cluster?
+
+No
+
+### Scalability
+
+###### Will enabling / using this feature result in any new API calls?
+
+No
+
+###### Will enabling / using this feature result in introducing new API types?
+
+No
+
+###### Will enabling / using this feature result in any new calls to the cloud provider?
+
+No
+
+###### Will enabling / using this feature result in increasing size or count of the existing API objects?
+
+Not immediately, but custom resource definitions might become larger (anticipating <10% size increase based on similar
+functionality in other systems).
+
+###### Will enabling / using this feature result in increasing time taken by any operations covered by existing SLIs/SLOs?
+
+Yes, custom resource create/update/patch/apply latencies will be impacted when the feature is used. We expect this to be negligible
+but will measure it before Beta.
+
+###### Will enabling / using this feature result in non-negligible increase of resource usage (CPU, RAM, disk, IO, ...) in any components?
+
+We don't expect it to. We will measure this before Beta.
```

Comment:
specifically, it would be good to demonstrate an upper bound on the resource cost someone could incur with a large+compact+complex cel expression vs existing large+compact+complex schema+validation rules

Reply:
Added